packages/lifecycle/src/runLifecycleHook.ts

```diff
@@ -91,7 +91,7 @@ export default async function runLifecycleHook (
 
 function getId (manifest: ImporterManifest | DependencyManifest) {
   if (!manifest.name) {
-    return undefined
+    return ''
```

Comment:
I think if this is not covered with a test, at least add this comment

```suggestion
    // Using an empty string instead of undefined or null
    // because this gets printed to the console
    return ''
```